### ISSUE TITLE
[Flake] push fails followed by push --config and then URL creation

### ISSUE THREAD

**Issue by amitkrout** [openshift/odo#1832]
[kind/bug]

## What versions of software are you using?
- Operating System: Supported platform
- Output of `odo version`: master

## How did you run odo exactly?
Check ```Any logs, error output, etc?``` tab

## Actual behavior
Push fails 

## Expected behavior
Push should pass

## Any logs, error output, etc?
```
Running odo with args: [create php testcmp --app e2e-xyzk --project nylolzxmxq --git https://github.com/appuio/example-php-sti-helloworld] 
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [2562047h47m16.854775807s]
[odo] Please use `odo push` command to create the component with source deployed
Running odo with args: [config set Ports 8080/TCP] 
[odo] Local config was successfully updated.
[odo] Run `odo push --config` command to apply changes to the cluster.
Running odo with args: [push --config] 
[odo] Validation
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [2562047h47m16.854775807s]
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [2562047h47m16.854775807s]
[odo] 
[odo] Configuration changes
[odo]  •  Creating component  ...
[odo]  ✓  Initializing component
[odo]  •  Triggering build from git  ...
[odo]  ✓  Triggering build from git [2562047h47m16.854775807s]
[odo]  •  Waiting for build to finish  ...
[odo]  ✓  Waiting for build to finish [2562047h47m16.854775807s]
[odo]  •  Deploying component testcmp  ...
[odo]  ✓  Deploying component testcmp [2562047h47m16.854775807s]
[odo]  ✓  Creating component [2562047h47m16.854775807s]
[odo]  •  Applying configuration  ...
[odo]  ✓  Applying configuration [2562047h47m16.854775807s]
Running odo with args: [url create myurl --port 8080] 
[odo]  ✓  URL myurl created
[odo] 
[odo] Run `odo push` to apply URL: myurl
Running odo with args: [push] 
[odo] Validation
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [2562047h47m16.854775807s]
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [2562047h47m16.854775807s]
[odo] 
[odo] Configuration changes
[odo]  •  Applying configuration  ...
[odo]  •  Triggering build from git  ...
[odo]  ✓  Triggering build from git [2562047h47m16.854775807s]
[odo]  •  Waiting for build to finish  ...
[odo]  ✓  Waiting for build to finish [2562047h47m16.854775807s]
[odo]  ✗  Failed to update config to component deployed
[odo]  ✗  Applying configuration [2562047h47m16.854775807s]
```

**Comment by amitkrout** [comment]
ping @kadel @mik-dass

**Comment by amitkrout** [comment]
Similar issue can be found slightly in a different scenario

```
odo generic creating component with an application and url 
  should create the component in default application
  /home/travis/gopath/src/github.com/openshift/odo/tests/integration/generic_test.go:58
Creating a new project: xhfszjwtyn
Running odo with args: [project create xhfszjwtyn -w -v4] 
[odo] I0702 15:31:31.909673   17490 preference.go:116] The configFile is /home/travis/.odo/preference.yaml
[odo] I0702 15:31:31.909715   17490 occlient.go:455] Trying to connect to server 127.0.0.1:8443
[odo] I0702 15:31:31.910152   17490 occlient.go:462] Server https://127.0.0.1:8443 is up
[odo] I0702 15:31:31.927287   17490 occlient.go:385] isLoggedIn err:  <nil> 
[odo]  output: "developer"
[odo] I0702 15:31:31.927320   17490 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: context
[odo] I0702 15:31:31.927329   17490 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: context
[odo] I0702 15:31:31.927362   17490 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: app
[odo] I0702 15:31:31.927370   17490 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: project
[odo] I0702 15:31:31.937726   17490 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: app
[odo] I0702 15:31:31.937844   17490 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: output
[odo] I0702 15:31:31.937867   17490 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: component
[odo]  •  Waiting for project to come up  ...
[odo]  ✓  Waiting for project to come up [473ms]
[odo]  ✓  Project 'xhfszjwtyn' is ready for use
[odo] I0702 15:31:32.434213   17490 odo.go:70] Could not get the latest release information in time. Never mind, exiting gracefully :)
[odo]  ✓  New project created and now using project : xhfszjwtyn
Created dir: /tmp/150033475
Current working dir: /home/travis/gopath/src/github.com/openshift/odo/tests/integration
Setting current dir to: /tmp/150033475
Running odo with args: [create php testcmp --app e2e-xyzk --project xhfszjwtyn --git https://github.com/appuio/example-php-sti-helloworld] 
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [471ms]
[odo] Please use `odo push` command to create the component with source deployed
Running odo with args: [config set Ports 8080/TCP] 
[odo] Local config was successfully updated.
[odo] Run `odo push --config` command to apply changes to the cluster.
Running odo with args: [push --config] 
[odo] Validation
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [244ms]
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [12ms]
[odo] 
[odo] Configuration changes
[odo]  •  Creating component  ...
[odo]  ✓  Initializing component
[odo]  •  Triggering build from git  ...
[odo]  ✓  Triggering build from git [547ms]
[odo]  •  Waiting for build to finish  ...
[odo]  ✓  Waiting for build to finish [2m]
[odo]  •  Deploying component testcmp  ...
[odo]  ✓  Deploying component testcmp [7s]
[odo]  ✓  Creating component [2m]
[odo]  •  Applying configuration  ...
[odo]  ✓  Applying configuration [11311ns]
Running odo with args: [push] 
[odo] Validation
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [737ms]
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [330ms]
[odo] 
[odo] Configuration changes
[odo]  •  Applying configuration  ...
[odo]  •  Triggering build from git  ...
[odo]  ✓  Triggering build from git [123ms]
[odo]  •  Waiting for build to finish  ...
[odo]  ✓  Waiting for build to finish [33s]
[odo]  ✗  Applying configuration [34s]
[odo]  ✗  Failed To Update Config To Component Deployed
Deleting project: xhfszjwtyn
Running odo with args: [project delete xhfszjwtyn -f] 
[odo] This project contains the following applications, which will be deleted
[odo] Application e2e-xyzk
[odo] This application has following components that will be deleted
[odo] component named testcmp
[odo] No services / could not get services
[odo]  •  Deleting project xhfszjwtyn  ...
[odo]  ✓  Deleting project xhfszjwtyn [8s]
[odo] Deleted project : xhfszjwtyn
Deleting dir: /tmp/150033475
Setting current dir to: /home/travis/gopath/src/github.com/openshift/odo/tests/integration
• Failure [160.088 seconds]
odo generic
/home/travis/gopath/src/github.com/openshift/odo/tests/integration/generic_test.go:16
  creating component with an application and url
  /home/travis/gopath/src/github.com/openshift/odo/tests/integration/generic_test.go:43
    should create the component in default application [It]
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/generic_test.go:58
    No future change is possible.  Bailing out early after 35.637s.
    Expected
        <int>: 1
    to match exit code:
        <int>: 0
    /home/travis/gopath/src/github.com/openshift/odo/tests/helper/helper_run.go:29
```
More details - https://travis-ci.com/openshift/odo/jobs/212660904

**Comment by mik-dass** [comment]
this seems to be related to #1782

**Comment by amitkrout** [comment]
Similar issue of almost same scenario found in url test file 
```
odoURLIntegration Listing urls 
  should list appropriate URLs and push message
  /go/src/github.com/openshift/odo/tests/integration/url_test.go:40

Created dir: /tmp/851661025
Creating a new project: aiiikqoxbb
Running odo with args: [project create aiiikqoxbb -w -v4] 
[odo] I0704 09:00:20.952251   28627 preference.go:116] The configFile is /tmp/851661025/config.yaml
[odo] I0704 09:00:20.952352   28627 occlient.go:455] Trying to connect to server api.ci-op-17tip006-f09f4.origin-ci-int-aws.dev.rhcloud.com:6443
[odo] I0704 09:00:21.202273   28627 occlient.go:462] Server https://api.ci-op-17tip006-f09f4.origin-ci-int-aws.dev.rhcloud.com:6443 is up
[odo] I0704 09:00:21.290529   28627 occlient.go:385] isLoggedIn err:  <nil> 
[odo]  output: "developer"
[odo] I0704 09:00:21.290571   28627 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: context
[odo] I0704 09:00:21.290582   28627 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: context
[odo] I0704 09:00:21.291367   28627 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: app
[odo] I0704 09:00:21.291400   28627 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: project
[odo] I0704 09:00:21.306976   28627 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: app
[odo] I0704 09:00:21.307014   28627 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: output
[odo] I0704 09:00:21.307023   28627 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: component
[odo]  •  Waiting for project to come up  ...
[odo]  ✓  Waiting for project to come up [357ms]
[odo]  ✓  Project 'aiiikqoxbb' is ready for use
[odo] I0704 09:00:21.679903   28627 odo.go:70] Could not get the latest release information in time. Never mind, exiting gracefully :)
[odo]  ✓  New project created and now using project : aiiikqoxbb
Running odo with args: [create nodejs --context /tmp/851661025 --project aiiikqoxbb myajaz --ref master --git https://github.com/openshift/nodejs-ex] 
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [476ms]
[odo] Please use `odo push` command to create the component with source deployed
Running odo with args: [push --context /tmp/851661025] 
[odo] Validation
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [417ms]
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [25ms]
[odo] 
[odo] Configuration changes
[odo]  •  Creating component  ...
[odo]  ✓  Initializing component
[odo]  •  Triggering build from git  ...
[odo]  ✓  Triggering build from git [83ms]
[odo]  •  Waiting for build to finish  ...
[odo]  ✓  Waiting for build to finish [2m]
[odo]  •  Deploying component myajaz  ...
[odo]  ✓  Deploying component myajaz [19s]
[odo]  ✓  Creating component [2m]
[odo]  •  Applying configuration  ...
[odo]  ✓  Applying configuration [13722ns]
[odo] 
[odo] Pushing to component myajaz of type git
[odo]  ✓  Changes successfully pushed to component
Running odo with args: [url list --context /tmp/851661025] 
[odo]  ✗  no URLs found for component myajaz in application app
Running odo with args: [url create qukgy --port 8080 --context /tmp/851661025] 
[odo]  ✓  URL qukgy created for component: myajaz
[odo] 
[odo] To create URL on the OpenShift Cluster, please use `odo push` 
Running odo with args: [url list --context /tmp/851661025] 
[odo] Found the following URLs for component myajaz in application app:
[odo] NAME      IN CONFIG     URL                          PORT
[odo] qukgy     Present       <not created on cluster>     8080
[odo] To create URLs on the OpenShift Cluster, please use `odo push` 
Running odo with args: [push --context /tmp/851661025] 
[odo] Validation
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [546ms]
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [239ms]
[odo] 
[odo] Configuration changes
[odo]  •  Applying configuration  ...
[odo]  •  Triggering build from git  ...
[odo]  ✓  Triggering build from git [82ms]
[odo]  •  Waiting for build to finish  ...
[odo]  ✓  Waiting for build to finish [2m]
[odo]  ✓  Applying configuration [2m]
[odo] 
[odo] Applying URL changes
[odo]  ✓  URL qukgy: http://qukgy-app-aiiikqoxbb.apps.ci-op-17tip006-f09f4.origin-ci-int-aws.dev.rhcloud.com created
[odo] 
[odo] Pushing to component myajaz of type git
[odo]  ✓  Changes successfully pushed to component
Running odo with args: [url list --context /tmp/851661025] 
[odo] Found the following URLs for component myajaz in application app:
[odo] NAME      IN CONFIG     URL                                                                                         PORT
[odo] qukgy     Present       http://qukgy-app-aiiikqoxbb.apps.ci-op-17tip006-f09f4.origin-ci-int-aws.dev.rhcloud.com     8080
Running odo with args: [url delete qukgy -f --context /tmp/851661025] 
[odo]  ✓  URL qukgy removed from the config file
[odo] To delete URL on the OpenShift Cluster, please use `odo push` 
Running odo with args: [url list --context /tmp/851661025] 
[odo] Found the following URLs for component myajaz in application app:
[odo] NAME      IN CONFIG     URL                                                                                         PORT
[odo] qukgy     Absent        http://qukgy-app-aiiikqoxbb.apps.ci-op-17tip006-f09f4.origin-ci-int-aws.dev.rhcloud.com     8080
[odo] To delete URLs on the OpenShift Cluster, please use `odo push` 
Running odo with args: [url create reabe --port 8000 --context /tmp/851661025] 
[odo]  ✓  URL reabe created for component: myajaz
[odo] 
[odo] To create URL on the OpenShift Cluster, please use `odo push` 
Running odo with args: [url list --context /tmp/851661025] 
[odo] Found the following URLs for component myajaz in application app:
[odo] NAME      IN CONFIG     URL                                                                                         PORT
[odo] reabe     Present       <not created on cluster>                                                                    8000
[odo] qukgy     Absent        http://qukgy-app-aiiikqoxbb.apps.ci-op-17tip006-f09f4.origin-ci-int-aws.dev.rhcloud.com     8080
[odo] To create/delete URLs on the OpenShift Cluster, please use `odo push` 
Running odo with args: [push --context /tmp/851661025] 
[odo] Validation
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [549ms]
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [239ms]
[odo] 
[odo] Configuration changes
[odo]  •  Applying configuration  ...
[odo]  •  Triggering build from git  ...
[odo]  ✓  Triggering build from git [73ms]
[odo]  •  Waiting for build to finish  ...
[odo]  ✓  Waiting for build to finish [1m]
[odo]  ✗  Failed To Update Config To Component Deployed
[odo]  ✗  Applying configuration [2m]
Deleting project: aiiikqoxbb
Running odo with args: [project delete aiiikqoxbb -f] 
[odo] This project contains the following applications, which will be deleted
[odo] Application app
[odo] This application has following components that will be deleted
[odo] component named myajaz
[odo] This component has following urls that will be deleted with component
[odo] URL named qukgy with host qukgy-app-aiiikqoxbb.apps.ci-op-17tip006-f09f4.origin-ci-int-aws.dev.rhcloud.com having protocol http at port 8080
[odo] No services / could not get services
[odo]  •  Deleting project aiiikqoxbb  ...
[odo]  ✓  Deleting project aiiikqoxbb [5s]
[odo] Deleted project : aiiikqoxbb
Deleting dir: /tmp/851661025


• Failure [349.392 seconds]
odoURLIntegration
/go/src/github.com/openshift/odo/tests/integration/url_test.go:13
  Listing urls
  /go/src/github.com/openshift/odo/tests/integration/url_test.go:39
    should list appropriate URLs and push message [It]
    /go/src/github.com/openshift/odo/tests/integration/url_test.go:40

    No future change is possible.  Bailing out early after 92.201s.
    Expected
        <int>: 1
    to match exit code:
        <int>: 0

    /go/src/github.com/openshift/odo/tests/helper/helper_run.go:29
```

**Comment by surajnarwade** [comment]
I am not able to produce this locally

**Comment by amitkrout** [comment]
> I am not able to produce this locally

@surajnarwade Give some more continuous try. Out of 10 you will hit this issue atleast 40% i guess ;)

**Comment by amitkrout** [comment]
Reproduced in CI - https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/1954/pull-ci-openshift-odo-master-integration/1061
```
odoURLIntegration Listing urls 
  should list appropriate URLs and push message
  /go/src/github.com/openshift/odo/tests/integration/url_test.go:41
 Created dir: /tmp/475069174
Creating a new project: qllqkgdnzd
Running odo with args: [project create qllqkgdnzd -w -v4] 
[odo] I0730 05:19:17.632597   31253 preference.go:116] The configFile is /tmp/475069174/config.yaml
[odo] I0730 05:19:17.632680   31253 occlient.go:486] Trying to connect to server api.ci-op-mvjt0q5q-f09f4.origin-ci-int-aws.dev.rhcloud.com:6443
[odo] I0730 05:19:17.651839   31253 occlient.go:493] Server https://api.ci-op-mvjt0q5q-f09f4.origin-ci-int-aws.dev.rhcloud.com:6443 is up
[odo] I0730 05:19:17.754536   31253 occlient.go:416] isLoggedIn err:  <nil> 
[odo]  output: "developer"
[odo] I0730 05:19:17.754579   31253 context.go:367] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: context
[odo] I0730 05:19:17.754591   31253 context.go:367] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: context
[odo] I0730 05:19:17.754648   31253 context.go:367] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: app
[odo] I0730 05:19:17.754663   31253 context.go:367] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: project
[odo] I0730 05:19:17.789847   31253 context.go:367] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: app
[odo] I0730 05:19:17.789873   31253 context.go:367] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: output
[odo] I0730 05:19:17.789878   31253 context.go:367] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: component
[odo]  •  Waiting for project to come up  ...
[odo]  ✓  Waiting for project to come up [492ms]
[odo]  ✓  Project 'qllqkgdnzd' is ready for use
[odo]  ✓  New project created and now using project : qllqkgdnzd
[odo] I0730 05:19:18.291503   31253 odo.go:70] Could not get the latest release information in time. Never mind, exiting gracefully :)
Running odo with args: [create nodejs --context /tmp/475069174 --project qllqkgdnzd zthnim --ref master --git https://github.com/openshift/nodejs-ex --port 8080,8000] 
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [61ms]
[odo] Please use `odo push` command to create the component with source deployed
[odo] 
Running odo with args: [push --context /tmp/475069174] 
[odo] Validation
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [100ms]
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [34ms]
[odo] 
[odo] Configuration changes
[odo]  •  Creating component  ...
[odo]  ✓  Initializing component
[odo]  •  Triggering build from git  ...
[odo]  ✓  Triggering build from git [121ms]
[odo]  •  Waiting for build to finish  ...
[odo]  ✓  Waiting for build to finish [2m]
[odo]  •  Deploying component zthnim  ...
[odo]  ✓  Deploying component zthnim [18s]
[odo]  ✓  Creating component [2m]
[odo]  •  Applying configuration  ...
[odo]  ✓  Applying configuration [12215ns]
[odo] 
[odo] Pushing to component zthnim of type git
[odo]  ✓  Changes successfully pushed to component
Running odo with args: [url list --context /tmp/475069174] 
[odo]  ✗  no URLs found for component zthnim in application app
Running odo with args: [url create laulo --port 8080 --context /tmp/475069174] 
[odo]  ✓  URL laulo created for component: zthnim
[odo] 
[odo] To create URL on the OpenShift Cluster, please use `odo push` 
Running odo with args: [url list --context /tmp/475069174] 
[odo] Found the following URLs for component zthnim in application app:
[odo] NAME      IN CONFIG     URL                          PORT
[odo] laulo     Present       <not created on cluster>     8080
[odo] To create URLs on the OpenShift Cluster, please use `odo push` 
Running odo with args: [push --context /tmp/475069174] 
[odo] Validation
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [199ms]
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [27ms]
[odo] 
[odo] Configuration changes
[odo]  •  Applying configuration  ...
[odo]  •  Triggering build from git  ...
[odo]  ✓  Triggering build from git [327ms]
[odo]  •  Waiting for build to finish  ...
[odo]  ✓  Waiting for build to finish [1m]
[odo]  ✗  Failed To Update Config To Component Deployed
[odo]  ✗  Applying configuration [1m]
Deleting project: qllqkgdnzd
Running odo with args: [project delete qllqkgdnzd -f] 
[odo] This project contains the following applications, which will be deleted
[odo] Application app
[odo] This application has following components that will be deleted
[odo] component named zthnim
[odo] No services / could not get services
[odo]  •  Deleting project qllqkgdnzd  ...
[odo]  ✓  Deleting project qllqkgdnzd [6s]
[odo] Deleted project : qllqkgdnzd
Deleting dir: /tmp/475069174
 • Failure [206.767 seconds]
odoURLIntegration
/go/src/github.com/openshift/odo/tests/integration/url_test.go:13
  Listing urls
  /go/src/github.com/openshift/odo/tests/integration/url_test.go:40
    should list appropriate URLs and push message [It]
    /go/src/github.com/openshift/odo/tests/integration/url_test.go:41
     No future change is possible.  Bailing out early after 87.915s.
      
    Expected
        <int>: 1
    to match exit code:
        <int>: 0
     /go/src/github.com/openshift/odo/tests/helper/helper_run.go:29
```

**Comment by amitkrout** [comment]
> I am not able to produce this locally

@surajnarwade I tried it locally too and unable to reproduce it on 3.11 cluster (No of try - 5). However this issue is inconsistently hit in CI.

My thought:
1. This may happen due to slow cluster 
2. Or probably due to some network glitch 
3. May be some other accept need to be verified

@surajnarwade Please share your thoughts cc_ @kadel

**Comment by amitkrout** [comment]
fixed via #2167
